content/200-concepts/100-components/02-prisma-client/250-opentelemetry-tracing.mdx

```diff
@@ -0,0 +1,324 @@
+
+```js
+// Imports
+import { ExpressInstrumentation } from '@opentelemetry/instrumentation-express'
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
+
+// Register your auto-instrumentors
+registerInstrumentations({
+  tracerProvider: provider,
+  instrumentations: [
+    new HttpInstrumentation(),
+    new ExpressInstrumentation(),
+    new PrismaInstrumentation(),
+  ],
+})
+```
+
+For a full list of available instrumentation, take a look at the [OpenTelemetry Registry](https://opentelemetry.io/registry/?language=js&component=instrumentation).
+
+### Customizing resource attributes
+
+You can adjust how your application's traces are grouped by changing the resource attributes to be more specific to your application:
+
+```js
+const provider = new NodeTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'weblog',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+```
+
+There is an ongoing effort to standardize common resource attributes. Whenever possible, it's a good idea to follow the [standard attribute names](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/README.md).
+
+### Reducing performance impact
+
+If your application sends a large number of spans to a collector, this can have a significant performance impact. You can use the following approaches to reduce this impact:
+
+- [Use the BatchSpanProcessor](#send-traces-in-batches-using-the-batchspanprocessor)
+- [Send fewer spans to the collector](#send-fewer-spans-to-the-collector-via-sampling)
+
+#### Send traces in batches using the `BatchSpanProcessor`
+
+In a production environment, you can use OpenTelemetry's `BatchSpanProcessor` to send the spans to a collector in batches rather than one at a time. However, during development and testing, you might not want to send spans in batches. In this situation, you might prefer to use the `SimpleSpanProcessor`.
+
+You can configure your tracing configuration to use the appropriate span processor, depending on the environment, as follows:
+
+```ts
+import {
+  SimpleSpanProcessor,
+  BatchSpanProcessor,
+} from '@opentelemetry/sdk-trace-base'
+
+if (process.env.NODE_ENV === 'production') {
+  provider.addSpanProcessor(new BatchSpanProcessor(otlpTraceExporter))
+} else {
+  provider.addSpanProcessor(new SimpleSpanProcessor(otlpTraceExporter))
+}
+```
+
+#### Send fewer spans to the collector via sampling
+
+Another way to reduce the performance impact is to [use probability sampling](https://opentelemetry.io/docs/reference/specification/trace/tracestate-probability-sampling/) to send fewer spans to the collector. This reduces the collection cost of tracing but still gives a good representation of what is happening in your application.
```

Comment:
Have we tested if this works as expected?

Reply:
Turns out no, but we have a workaround to make it work in the meantime. So all good, we can keep it in the docs.